src/assets/src/components/addQueue.tsx

```diff
@@ -0,0 +1,388 @@
+    ): Promise<QueueHost> => {
+        return await api.createQueue(queueName, allowedBackends, queueDescription, hosts);
+    };
+    const [doAddQueue, addQueueLoading, addQueueError] = usePromise(
+        addQueue,
+        (queue: QueueHost) => {
+            recordQueueManagementEvent('Added Queue');
+            queue.hosts.map(h => recordQueueManagementEvent('Added Host'));
+            location.href = `/manage/${queue.id}/`;
+        }
+    );
+
+    // Validation functions
+    type ValidationResultSetter = React.Dispatch<React.SetStateAction<undefined | ValidationResult>>;
+    function validateSomething (someValue: string, schema: StringSchema, someSetter: ValidationResultSetter): ValidationResult {
+        const validationResult = validateString(someValue, schema, true);
+        someSetter(validationResult);
+        return validationResult;
+    };
+    const validateMeetingTypes = (allowedBackends: Set<string>): boolean => {
+        const isInvalid = allowedBackends.size === 0;
+        setAllowedIsInvalid(isInvalid);
+        return isInvalid;
+    };
+
+    // On change handlers
+    const handleNameChange = (newName: string) => {
+        setName(newName);
+        validateSomething(newName, queueNameSchema, setNameValidationResult);
+    };
+    const handleDescriptionChange = (newDescription: string) => {
+        setDescription(newDescription);
+        validateSomething(newDescription, queueDescriptSchema, setDescriptValidationResult);
+    };
+    const handleAllowedChange = (newAllowedBackends: Set<string>) => {
+        setAllowedMeetingTypes(newAllowedBackends);
+        validateMeetingTypes(newAllowedBackends);
+    };
+
+    // On click handlers
+    const handleGeneralNextClick = () => {
+        let curNameValidationResult = nameValidationResult;
+        let curDescriptValidationResult = descriptValidationResult;
+        let curAllowedIsInvalid = allowedIsInvalid;
+        if (!nameValidationResult) {
+            curNameValidationResult = validateSomething(name, queueNameSchema, setNameValidationResult);
+        }
+        if (!descriptValidationResult) {
+            curDescriptValidationResult = validateSomething(description, queueDescriptSchema, setDescriptValidationResult);
+        }
+        if (allowedIsInvalid === undefined) {
+            curAllowedIsInvalid = validateMeetingTypes(allowedMeetingTypes);
+        }
+        if (!curNameValidationResult!.isInvalid && !curDescriptValidationResult!.isInvalid && curAllowedIsInvalid === false) {
+            setActiveKey('hosts');
+        }
+    };
+
+    // Improve this?
+    const handleManageHostsFinishClick = () => {
+        if (name !== '' && allowedMeetingTypes.size !== 0) {
+            doAddQueue(name, allowedMeetingTypes, description, hosts);
+        } else {
+            throw Error('Attempted to pass invalid data to API for queue creation')
+        }
+    };
```

Comment:
This check could probably be cleaned up. I don't think it's possible because of validation (but could be wrong of course) to reach this point with invalid data (i.e. an empty string for a name and no selected meeting types).

Reply:
Yeah, I could go either way on this. It will catch errors if we have a regression with validation in the future. OTOH, if code isn't used, I will generally remove it just to minimize cognitive overhead.

Reply:
I guess I'll leave this alone for now. I may return to it in the future.